libraries/AP_Common/AP_Common.cpp

```diff
@@ -70,3 +70,16 @@ bool hex_to_uint8(uint8_t a, uint8_t &res)
     }
     return true;
 }
+
+/*
+  strncpy without the warning for not leaving room for nul termination
+ */
+void strncpy_noterm(char *dest, const char *src, size_t n)
```

Comment:
Shouldn't this really add the null as strncpy usually does - if there's room?

Reply:
it does

Reply:
it does